repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -195,6 +201,67 @@ public AtlasLineageOnDemandInfo getAtlasLineageInfo(String guid, LineageOnDemand
+    @GraphTransaction
+    public AtlasLineageListInfo getAtlasLineageListInfo(String guid, LineageListRequest lineageListRequest) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("getAtlasListInfo");
+
+        RequestContext.get().setRelationAttrsForSearch(lineageListRequest.getRelationAttributes());
+
+        AtlasLineageListInfo ret;
+
+        AtlasLineageListContext atlasLineageOnDemandContext = new AtlasLineageListContext(lineageListRequest, atlasTypeRegistry);
+
+        boolean isDataSet = validateEntityTypeAndCheckIfDataSet(guid);
+
+        ret = getLineageListInfoOnDemand(guid, atlasLineageOnDemandContext, isDataSet);
+
+        ret.setSearchParameters(lineageListRequest);
+
+        scrubLineageEntities(ret.getEntities());
+
+        RequestContext.get().endMetricRecord(metricRecorder);
+
+        return ret;
+    }
+
+    @Override
+    @GraphTransaction
+    public AtlasLineageSizeInfo getAtlasLineageSize(LineageSizeRequest lineageSizeRequest) throws AtlasBaseException {
+        AtlasPerfMetrics.MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("getAtlasLineageSize");
+        final String guid = lineageSizeRequest.getGuid();
+        boolean isDataSet = validateEntityTypeAndCheckIfDataSet(guid);
+        final AtlasLineageSizeInfo ret = getLineageSizeForDirection(isDataSet, new AtlasLineageSizeContext(lineageSizeRequest, atlasTypeRegistry));
+        ret.setSearchParameters(lineageSizeRequest);
+        RequestContext.get().endMetricRecord(metricRecorder);
+        return ret;
+    }
+
+    private AtlasLineageSizeInfo getLineageSizeForDirection(boolean isDataSet, AtlasLineageSizeContext atlasLineageSizeContext) {
+        AtomicInteger count = new AtomicInteger(0);
+        int currentDepth = 0;
+        boolean isLimitReached = false;
+        if (isDataSet) {
+            AtlasVertex datasetVertex = AtlasGraphUtilsV2.findByGuid(this.graph, atlasLineageSizeContext.getGuid());
+            if (atlasLineageSizeContext.getDirection() == LineageSizeRequest.LineageDirection.INPUT) {
+                isLimitReached = calculateLineageSize(datasetVertex, true, atlasLineageSizeContext, currentDepth, count);
+            } else if (atlasLineageSizeContext.getDirection() == LineageSizeRequest.LineageDirection.OUTPUT) {
+                isLimitReached = calculateLineageSize(datasetVertex, false, atlasLineageSizeContext, currentDepth, count);
+            }
+        } else {
+            AtlasVertex processVertex = AtlasGraphUtilsV2.findByGuid(this.graph, atlasLineageSizeContext.getGuid());
+            // make one hop to the next dataset vertices from process vertex and traverse with 'depth = depth - 1'
+            if (atlasLineageSizeContext.getDirection() == LineageSizeRequest.LineageDirection.INPUT) {
+                Iterator<AtlasEdge> processEdges = processVertex.getEdges(AtlasEdgeDirection.OUT, PROCESS_INPUTS_EDGE).iterator();
```

Comment:
shouldnt this be INPUT edges ? as we are processing lineage for INPUT

Reply:
This should be Output edges only, the direction from Process is always Output irrespective of input or output edges. You can refer to Process atlas model for understanding.